### DISCUSSION TITLE
Политики гитхаба

### DISCUSSION THREAD

**Discussion by glitchcore** [b4ck5p4c3/0x08.in#363] [General]
Нигде не нашел правил работы с нашим основным репозиторием.

1. Есть ли какая-то политика названия веток?
2. Какой основной flow для добавления изменений, я вижу что существующие PR вливаются в dev. А когда и по каким правилам dev вливается в master? Если я хочу создать новую ветку, нужно ветвиться от dev или от master? То, что лежит на 0x08.in — это master?

Я бы хотел поднять вопрос возврата к старой модели, когда PR не требовал аппрува, вливался через сутки или по дедлайну, и можно было отложить вливание при наличии возражений.

С одной стороны, наблюдаю ситуацию, когда несколько PR висят в ожидании аппрува и авторам приходится бегать по хакспейсу и просить ревью. Могу представить себе вопрос, который принципиален для 1-2 человек, тогда придется искать третьего человека «для галочки».
С другой стороны, наблюдал как некоторые правки вливались за пару минут, получая аппрув от пары участников. В такой ситуации даже если у кого-то возникнут возражения, им придется откатывать изменения. Для некоторых участников гит не является ежедневным инструментом, и эта процедура становится сильно сложнее чем обсудить висящий PR, и может создать ситуацию, когда несколько человек проталкивают изменения, другим участникам это не нравится, но они не находят сил с этим что-то сделать, или игнорируют изменения в правилах.

**Comment by glitchcore** [comment]
Напоминаю, старые правила ведения PR и issues находятся на сайте: https://0x08.in/wiki/issue-rules. Абзац про «Команды в PR и issues» можно считать устаревшим, т.к. автоматизацию так и не накрутили, хотя этим пытались пользоваться в ручном режиме

**Comment by heeeyflo** [comment]
Есть контрибьютинг-гайдлайны по эвентам и проектам:
 * https://github.com/b4ck5p4c3/0x08.in/tree/dev/content/events
 * https://github.com/b4ck5p4c3/0x08.in/tree/dev/content/projects

Я согласен с аргументами @glitchcore и механизм апррува мне тоже кажется избыточным и не [реализующим целей](https://github.com/b4ck5p4c3/0x08.in/issues/342#issuecomment-758191134), описанных Лёней.

**Reply by glitchcore** [reply]
О, спасибо за гайдлайны, мб стоит засунуть ссылку на них в вики

**Comment by glitchcore** [comment]
Возвращаясь к аргументам за подход с аппрувами:

> 1. Обмен экспертизой и вовлечение разных участников к изменениям на сайте.

Судя по предыдущим PR, созданным после изменения политики, аппрувят те, кто и так участвовал в обсуждениях и написании PR. Я думаю, что вопрос обмена экспертизой лучше решать более гибко: если у человека есть экспертиза и он готов потратить время на ревью или обсуждение, пусть он делает это по собственной воле. Всегда можно спросить конкретного участника о его мнении, если вы создали или увидели релевантный PR. Мне не нравится, что политика _заставляет людей участвовать в жизни хакспейса_, не думаю, что это повышает качество такого участия.
Еще один очень важный момент: сейчас правила говорят, что достаточно двух аппрувов, т.е. два участника, не обладающие экспертизой, но которые постоянно тусят в GH, могут ткнуть аппрув (с добрыми намерениями, чтобы в хакспейсе дела двигались), и даже если был человек с экспертизой, вопрос решится без него. Таким образом, вовлечение разных участников скорее не происходит.

> 2. Осведомленность. Больше внимания к изменениям – больше людей вкурсе, что происходит.

Я считаю, что привлекать внимание к изменениям можно, уведомляя об этих изменениях в канале. + те кто подписан, получают уведомления на почту. Считаю, что привлечение внимания через необходимость ткнуть галочку уменьшает ценность аппрувов.

> 3. Коллективное редактирование и модерация. Меньше ошибок, меньше шансов залить код/содержимое противоречащее правилам, принципам, целям, задачам.

Опять же, раньше были сутки на то чтобы много людей могло вычитать PR, сейчас PR может быть принят за 10 минут. Т.е. мы заменяем коллектив из 20+ человек на коллектив из 3-х. Даже если ревьюверы внимательно прочитали текст и добросовестно аппрувнули, у них может не быть какой-то информации, которая есть у других

**Comment by sntxerror** [comment]
Если это будет 1 аппрув + сутки и отсутствие возражений?

**Reply by glitchcore** [reply]
По какому негативному сценарию будут развиваться события, если аппрувы убрать вовсе?
